apps/src/blockly/addons/workspaceSvgFrame.js

```diff
@@ -108,10 +108,16 @@ export default class WorkspaceSvgFrame extends SvgFrame {
 
   getFrameX() {
     // In LTR the svg should be to the right of the toolbox, plus a margin.
+    const metricsManager = this.element_.getMetricsManager();
     let frameX = frameSizes.MARGIN_SIDE / 2;
-    const toolbox = this.element_.toolbox_;
-    if (toolbox) {
-      frameX += toolbox.width_;
+    // Toolbox width > 0 if we have a categorized toolbox.
+    const toolboxWidth = metricsManager.getToolboxMetrics().width;
```

Comment:
Any chance for either metrics object to be undefined?

Reply:
Per their docs [1](https://developers.google.com/blockly/reference/js/blockly.metricsmanager_class.getflyoutmetrics_1_method.md), [2](https://developers.google.com/blockly/reference/js/blockly.metricsmanager_class.gettoolboxmetrics_1_method.md), can only return `ToolboxMetrics`, not null or undefined.